lib/src/components/NxTextInput/NxTextInput.tsx

```diff
@@ -48,6 +49,7 @@ export const PrivateNxTextInput = forwardRef<HTMLDivElement, Props>(
       const isTextArea = type === 'textarea',
           element = isTextArea ? 'textarea' : 'input',
           typeAttr = isTextArea ? undefined : (type || 'text'),
+          classes = rows && isTextArea ? 'nx-text-input__input textarea-custom-sizing' : 'nx-text-input__input',
```

Comment:
Use the `classnames` utility for conditional classes.  See a few lines below this for an example.